### PR TITLE
Delete Buttons - Comment

### PR DESCRIPTION
**Pull request by Khilio1081**
**OVERVIEW**
- Added delete button inside the comments modal.

**SNIPPET**
image

### CODE REVIEW

File: src/app/Http/Controllers/PostCommentController.php [hunk 1]
```diff
@@ -2,44 +2,47 @@
...
     {
-        $this->postCommentService = new PostCommentService();
+        $this->postCommentService = $postCommentService;
     }
 
-    /**
-     * Calls service to create a new comment.
-     */
-    public function create(PostCommentRequest $request):RedirectResponse
+    public function create(Request $request): RedirectResponse
```
Comment (hipe01mark) [inline src/app/Http/Controllers/PostCommentController.php:23]: docbloc

File: src/app/Http/Controllers/PostCommentController.php [hunk 2]
```diff
@@ -2,44 +2,47 @@
...
     }
 
-    /**
-     * Calls service to update comment.
-     */
-    public function update(PostCommentRequest $request):RedirectResponse
-    {   
-        $this->postCommentService->update($request);
-        return redirect('/posts-page');
+    public function delete(Request $request): RedirectResponse
```
Comment (hipe01mark) [inline src/app/Http/Controllers/PostCommentController.php:33]: docbloc

File: src/app/Http/Controllers/PostCommentController.php [hunk 3]
```diff
@@ -2,44 +2,47 @@
...
+    {
+        $this->postCommentService->delete($request->postCommentToDeleteId);
+        return redirect()->back();
     }
 
-    /**
-     * Calls service to delete a  comment.
-     */
-    public function delete(Request $request):RedirectResponse
+    public function update(Request $request, $id): RedirectResponse
```
Comment (hipe01mark) [inline src/app/Http/Controllers/PostCommentController.php:39]: docbloc

File: src/app/Services/PostCommentService.php [hunk 4]
```diff
@@ -2,16 +2,14 @@
...
+use Illuminate\Http\Request;
 
 class PostCommentService
 {
-    /**
-     * Creates a new comment for the post.
-     */
-    public function create(PostCommentRequest $request): void
-    {   
+    public function create(Request $request): void
```
Comment (hipe01mark) [inline src/app/Services/PostCommentService.php:11]: docbloc

File: src/app/Services/PostCommentService.php [hunk 5]
```diff
@@ -20,19 +18,14 @@ public function create(PostCommentRequest $request): void
         ]);
     }
 
-    /**
-     * Edits comment for the post.
-     */
-    public function update(PostCommentRequest $request): void
+    public function delete(int $postCommentToDeleteId): void
```
Comment (hipe01mark) [inline src/app/Services/PostCommentService.php:21]: docbloc

File: src/app/Services/PostCommentService.php [hunk 6]
```diff
@@ -20,19 +18,14 @@ public function create(PostCommentRequest $request): void
...
     {
-        PostComment::findOrFail($request->postCommentToEditId)->update(['content' => $request->content]);
+        PostComment::destroy($postCommentToDeleteId);
     }
 
-    /**
-     * Deletes a comment.
-     */
-    public function delete(int $postCommentToDeleteId): void 
+    public function update(Request $request, int $id): void
```
Comment (hipe01mark) [inline src/app/Services/PostCommentService.php:26]: docbloc

File: src/resources/views/components/sections/comment-section.blade.php [hunk 7]
```diff
@@ -1,23 +1,50 @@
...
+                        <form method="POST" action="{{ route('post.comment.delete', ['id' => $comment->id]) }}" class="delete-comment-form inline">
+                            @csrf
+                            @method('DELETE')
+                            <input type="hidden" name="postCommentToDeleteId" value="{{ $comment->id }}">
+                            <button type="submit" class="text-red-500 text-sm bg-transparent border-none cursor-pointer">
+                                Delete
+                            </button>
+                        </form>
+                    </div>
+                    <!-- Edit Comment Form -->
```
Comment (hipe01mark) [inline src/resources/views/components/sections/comment-section.blade.php:33]: remove

File: src/app/Http/Controllers/PostCommentController.php [hunk 8]
```diff
@@ -2,44 +2,47 @@
 
 namespace App\Http\Controllers;
 
-use Illuminate\Http\Request;
+use App\Http\Requests\PostCommentRequest;
+use App\Models\PostComment; // Ensure this is imported
+use App\Models\PostShare; // Ensure this is imported
```
Comment (mikco-hipe) [inline src/app/Http/Controllers/PostCommentController.php:7]: remove comments

**Review by mikco-hipe — COMMENTED** [review]
Please check!

**Pull request closed without merging**